dev/image-guidelines.md

```diff
@@ -0,0 +1,329 @@
+# Guidelines for creating image content
+
+This document provides guidance for creating image content for the Velociraptor documentation.
+It specifies some informal standards and gives some tool-specific advice with the goal of ensuring as much consistency as possible in our image content.
+
+---
+
+* [Image categories](#image-categories)
+* [Fonts](#fonts)
+* [Colors](#colors)
+* [Excalidraw object library](#excalidraw-object-library)
+* [Screenshots](#screenshots)
+   * [Screen size](#screen-size)
+      * [Partial screens](#partial-screens)
+      * [Full screens](#full-screens)
+   * [Velociraptor theme](#velociraptor-theme)
+      * [Theme tweaks](#theme-tweaks)
+   * [Annotations](#annotations)
+      * [Arrows](#arrows)
+      * [Numbered callouts](#numbered-callouts)
+      * [Excalidraw-specific options](#excalidraw-specific-options)
+* [Line diagrams](#line-diagrams)
+* [Exporting images](#exporting-images)
+   * [Formats and settings](#formats-and-settings)
+   * [Fonts in SVG images](#fonts-in-svg-images)
+   * [File size](#file-size)
+
+---
+
+## Image categories
+
+Image content generally falls into one of these categories:
+
+1. [Screenshots](#screenshots)
+	- unannotated
+	- annotated
+2. [Line diagrams](#line-diagrams) (flow charts, sequence diagrams, etc.)
+
+For **annotated screenshots** and **line diagrams** we use Excalidraw because it's easy to use and it produces drawings with a realistic hand-drawn effect.
+For annotated screenshots this aspect contrasts well with inorganic GUI elements, and in general it's humanistic and visually appealing.
+
+For **unannotated screenshots** we use the PNG format since it has lossless compression which preserves the legibility in text in images of the Velociraptor GUI. There's no need to do anything fancy with these images. Just try to keep the size below 300KB if possible.
+
+## Fonts
+
+The standard Excalidraw version uses a default font named Virgil which is too scrawly and therefore a bit hard to read (as discussed [here](https://github.com/excalidraw/excalidraw/issues/2945)).
+The default fonts are not able to be changed without editing the application's source, and the Excalidraw developers have stated disinterest in make custom fonts possible.
+
+Fortunately there exists a reasonably well-maintained fork which allows adding custom fonts
+:   https://github.com/hulkbig/excalidraw-zh
+
+It's very easy to set up a local instance:
+
+1. clone the repo and cd to it
+2. `yarn` to install
+3. `yarn start` to start the web app
+4. follow the link that it shows you (by default http://localhost:3000/).
+
+Only 3 font variants are supported so we assign them as follows:
```

Comment:
I like the idea of a consistent font everywhere - lets just coerce it with a presubmit - that way this docs are not relevant

Reply:
Yeah the font will be repositioned within the bounding box - but as long as the bounding box is reasonable it should flow within it.

Just tried it and it makes no practical difference to the font positioning:

Original font
![image](https://github.com/user-attachments/assets/42fd255a-6bf3-4a86-8b6c-96281f9cff26)

Replaced with our font
![image](https://github.com/user-attachments/assets/e34034e2-f76b-4799-b323-b81860d0762e)

I lean towards just making the instructions simpler. If a power user really wants to use the exact same font - go for it. But we dont need to mandate that in this style guide.

Reply:
The instructions are not going to be simpler if you have to explain that the user should imagine different fonts while drawing, other than the ones their eyes are seeing.

![query_api_client](https://github.com/user-attachments/assets/75941452-855b-4526-9f6c-d44ff91c98cc)

I've done this test with Excalidraw (the official version) and the bounding box resizes vertically and wraps the text. So that would require us to always add extra space to the bounding box so that it doesn't change size with the font swap. And then also try to explain why that's necessary and provide some estimation of how much extra space is sufficient.

![Screenshot from 2024-09-19 11-21-30](https://github.com/user-attachments/assets/cc8fa8e3-37d7-425c-ae35-e21af804ed66)

I would really like to use the official version (although not their hosted app which changes without notice) but they don't support custom fonts and their default fonts suck. If you read their issues board it's frustration that a lot of people have. They've added a font picker which hints at custom font becoming possible in future, but not yet: https://github.com/excalidraw/excalidraw/pull/8012

The official version is currently doing this with the SVG exports:
- allowing 4 default fonts (3 GUI presets + 1 extra)
- embedding them multiple times

![Screenshot from 2024-09-19 11-44-07](https://github.com/user-attachments/assets/70021269-0201-4009-be16-80c5d1d89daa)

We could swap a 4-font definitions section and map to those font names but what if they decide to add more or change the names of the existing ones? They're done that in the past (where "Virgil" became "Excalifont") and the change I linked to above suggests that the font aspect is currently very much in flux and undecided.